### ISSUE TITLE
Arguments of functions passed to TTree::Draw/Scan always take the default values

### ISSUE THREAD

**Issue by ahmarinojad** [root-project/root#14405]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

Dear experts,
since ROOT 6.30 I have encountered an issue when calling simple functions inside a `TTree::Scan`. For example, if I consider a function whose prototype is 
`int func(int a)`
and call it in a `TTree::Scan`
`t->Scan("func(myBranch)")`

it will always be called with 0 as argument. This happens also when passing a constant argument (`t->Scan("func(2)")`). This issue doesn't appear in root 6.28.

Cheers,
Marino

### Reproducer

To reproduce the bug just call the `test()` function in the following piece of code
```
root[] .L filename.C
root[] test()
```

If this macro is run with root 6.28, the `cout` in `testfunc` will correctly print the values of the branches and `testfunc` column in the Scan will show the correct output of the formula `a+b*c`. When run with ROOT 6.30, the `cout` will print `0 0 0` and the `testfunc` column will always be 0.

```
#include <iostream>
#include <string>
#include <TFile.h>
#include <TTree.h>

using namespace std;

float testfunc(int a, int b, float c)
{
    cout << a << " " << b << " " << c << endl;
    return a + b * c;
}

void createTree(const string &filename)
{
    cout << "Creating tree in file " << filename << endl;
    TFile *f = TFile::Open(filename.c_str(), "RECREATE");
    TTree *t = new TTree("t", "t");
    int a;
    int b;
    float c;
    t->Branch("a", &a, "a/I");
    t->Branch("b", &b, "b/I");
    t->Branch("c", &c, "c/F");
    for (int i = 0; i < 10; ++i)
    {
        a = i;
        b = i + 1;
        c = i + 2;
        t->Fill();
    }
    t->Write();
    f->Close();
    cout << "Tree created" << endl;
}

void runExampleScan(const string &filename)
{
    cout << "Running example scan" << endl;
    TFile *f = TFile::Open(filename.c_str());
    TTree *t = (TTree *)f->Get("t");
    t->Scan("a:b:c:testfunc(a,b,c)", "", "colsize=20");
    delete t;
    delete f;
}

void runExampleDraw(const string &filename)
{
    gROOT->SetBatch();
    TCanvas *c = new TCanvas("c", "c", 800, 600);
    cout << "Running example draw" << endl;
    TFile *f = TFile::Open(filename.c_str());
    TTree *t = (TTree *)f->Get("t");
    t->Draw("testfunc(a,b,c)", "");
    delete t;
    delete f;
    c->SaveAs("test.png");
}

int test()
{
    string filename = "test.root";
    createTree(filename);
    runExampleScan(filename);
    runExampleDraw(filename);
    return 0;
}

```

### ROOT version

```
ROOT v6.30/02
Built for linuxx8664gcc on Dec 20 2023, 18:59:44
From heads/master@tags/v6-30-02
With g++ (GCC) 13.1.0
Binary directory: /cvmfs/sft.cern.ch/lcg/releases/ROOT/6.30.02-fb5be/x86_64-el9-gcc13-opt/bin
```

### Installation method

Pre-built binary

### Operating system

Red Hat Enterprise Linux 9.3 (tested also on Centos7)

**Comment by vkucera** [comment]
Bisecting the issue located the origin to be between commits: ee112dfe1da04590a0fd8ed9656eb4a83a948596 (last good) and de19eb1ddb9d1b2edcb5d2187078b2d67bb7b784 (first bad), concluding that the bug was introduced with https://github.com/root-project/root/pull/11252 .

**Comment by vkucera** [comment]
Commit 9dd78b6ea3 is the culprit.

**Comment by ktf** [comment]
Just to add on this. Please, consider it an official ALICE request. Without this fixed we might have to revert back to 6.28.0.

**Comment by ktf** [comment]
Fixed by #14853, at least for us.
